pages/integrations/artifactory.md.erb

```diff
@@ -0,0 +1,119 @@
+      <em>Example:</em> <code>http://my-domain.com/artifactory</code><br>
+    </td>
+  </tr>
+  <tr>
+    <td><code>BUILDKITE_ARTIFACTORY_USER</code></td>
+    <td>
+      The username of a user configured in your Artifactory instance<br>
+      <em>Example:</em> <code>cdanvers</code><br>
+    </td>
+  </tr>
+  <tr>
+    <td><code>BUILDKITE_ARTIFACTORY_PASSWORD</code></td>
+    <td>
+      The password for your Artifactory user<br>
+      <em>Example:</em> <code>xxxx</code><br>
+    </td>
+  </tr>
+</table>
+
+_Optional vars:_
+
+<table>
+  <tr>
+    <td><code>BUILDKITE_ARTIFACT_UPLOAD_DESTINATION</code></td>
+    <td>
+      The path you'd like to upload to, including the rt:// prefix<br>
+      <em>Example:</em> <code>"rt://generic-repo/build-$BUILDKITE_BUILD_NUMBER/$BUILDKITE_JOB_ID/"</code><br>
+    </td>
+  </tr>
+</table>
+
+By default, any artifacts will be uploaded to the repository inside a folder named with the ID of the job that uploaded it. Specifying the `BUILDKITE_ARTIFACT_UPLOAD_DESTINATION` variable will overwrite this default. 
+
+For example, the below Agent environment hook is stored securely on the Agent machine and exports all of the available variables:
+
+```bash
+export BUILDKITE_ARTIFACTORY_URL=http://my-domain:8081/artifactory
+export BUILDKITE_ARTIFACTORY_USER=admin
+export BUILDKITE_ARTIFACTORY_PASSWORD=fishes-are-the-coolest
+export BUILDKITE_ARTIFACT_UPLOAD_DESTINATION="rt://generic-repo/$BUILDKITE_JOB_ID/"
+```
+
+For best practices and recommendations about secret storage in the Agent, see the [Managing Secrets](/docs/pipelines/secrets) guide.
+
+## Uploading and downloading artifacts
+
+Once you've configured the Artifactory `buildkite-agent` environment variables, the `artifact_paths` attribute of command steps will automatically use the Artifactory configuration to upload artifacts. 
+
+The following example `pipeline.yml` file will build a golang binary and upload it to Artifactory. The artifact will be visible in the build on Buildkite and can be downloaded in future steps using `buildkite-agent artifact download`.
+
+```yml
+steps:
+  - label: ":golang: :package:"
+    command: "go build -v -o myapp-darwin-amd64"
+    artifact_paths: "myapp-darwin-amd64"
+    plugins:
+      - docker#v3.3.0:
+          image: "golang:1.11"
+```
+
+
+
+You can also use [buildkite-agent artifact](https://buildkite.com/docs/agent/v3/cli-artifact) commands directly, for example `buildkite-agent artifact upload my-file` and `buildkite-agent artifact download my-file`.
+
+<%= image "buildkite-artifact-step.png", width: 2320/2, height: 822/2, alt: "Screenshot of a Buildkite command step's output logging an artifact upload to Artifactory" %>
+
+<%= image "artifactory-go-local-repository.png", width: 1486/2, height: 710/2, alt: "Screenshot of an artifact in the go-local repository in Artifactory" %>
+
+All artifacts in Buildkite link to their location in Artifactory. Click through to view them in Artifactory.
```

Comment:
@toolmantim is this true? The go binary I uploaded with the example code doesn't link through to Artifactory - if you click it, it just gets downloaded to your computer 🤔

Reply:
Hrmm, is there an `BUILDKITE_ARTIFACTORY_ACCESS_URL` setting similar to `BUILDKITE_S3_ACCESS_URL`?

Reply:
Not as far as I'm aware, I couldn't see one defined in the code alongside the other `BUILDKITE_ARTIFACTORY_*` variables. Should there be one?

Reply:
I think it's `BUILDKITE_S3_ACCESS_URL` that sets the URL endpoint if you want to be able to link to it directly from the Buildkite web UI?